src/common/webrequest_curl.cpp

```diff
@@ -296,15 +415,44 @@ void wxWebRequestCURL::HandleCompletion()
     }
 }
 
+bool wxWebRequestCURL::HasPendingCancel() const
+{
+    return m_cancelPending;
+}
+
+bool wxWebRequestCURL::HasInternalFail() const
+{
+    return m_hasInternalFail;
+}
+
+void wxWebRequestCURL::SetInternalFail(const wxString& msg)
+{
+    m_cancelPending = true;
+    m_hasInternalFail = true;
+    m_internalFailMsg = msg;
+}
+
 wxString wxWebRequestCURL::GetError() const
 {
-    // We don't know what encoding is used for libcurl errors, so do whatever
-    // is needed in order to interpret this data at least somehow.
-    return wxString(m_errorBuffer, wxConvWhateverWorks);
+    if ( m_hasInternalFail )
+    {
+        return m_internalFailMsg;
+    }
+    else
+    {
+        // We don't know what encoding is used for libcurl errors, so do
+        // whatever is needed in order to interpret this data at least somehow.
+        return wxString(m_errorBuffer, wxConvWhateverWorks);
+    }
 }
 
 size_t wxWebRequestCURL::CURLOnRead(char* buffer, size_t size)
 {
+    if ( HasPendingCancel() )
+    {
+        return ( size == 1 ) ? 0 : 1;
```

Comment:
I'd probably use a small inline helper `size_t CURLCallbackError(size_t size)` doing this just to make the code more self-documenting.

Reply:
This line has simply been removed since I'm no longer trying to use the callbacks to cancel the transfer.